File: 1402-reducing-dishes/1402-reducing-dishes.cs
public class Solution {
    public int MaxSatisfaction(int[] satisfaction) {
        Array.Sort(satisfaction);
        int n = satisfaction.Length;
        int res = 0, tmp = 0;
        for(int i = n - 1; i >= 0 && satisfaction[i] + tmp > 0; --i) {
            tmp += satisfaction[i];
            res += tmp;
        }
        return res;
    }
}

File: 1402-reducing-dishes/1402-reducing-dishes.js
/**
 * @param {number[]} satisfaction
 * @return {number}
 */
var maxSatisfaction = function(satisfaction) {
    satisfaction.sort((a, b) => a - b);
    console.log(satisfaction)
        let n = satisfaction.length;
        let res = 0, tmp = 0;

        for(let i = n - 1; i >= 0; --i) {
            if(satisfaction[i] + tmp <=  0) break;
            tmp += satisfaction[i];
            res += tmp;
        }
    return res;
};